pylint/checkers/similar.py

```diff
@@ -466,12 +809,16 @@ def get_map_data(self):
         """Passthru override"""
         return Similar.get_map_data(self)
 
-    @classmethod
-    def reduce_map_data(cls, linter, data):
+    def reduce_map_data(self, linter, data):
         """Reduces and recombines data into a format that we can report on
 
         The partner function of get_map_data()"""
         recombined = SimilarChecker(linter)
+        recombined.min_lines = self.min_lines
+        recombined.ignore_comments = self.ignore_comments
+        recombined.ignore_docstrings = self.ignore_docstrings
+        recombined.ignore_imports = self.ignore_imports
+        recombined.ignore_signatures = self.ignore_signatures
```

Comment:
Those lines are similar to MR #4175 except i had to add the ignore_* options to retrieve the same options in each process.

Reply:
I don't understand how they are usefull yet. I tried passing the tests without them and tests passes.

Reply:
If I understand correctly : all data received by reduce_map_data comes from get_map_data, which is `return self.linesets`. LineSets here is already "filtered" by the ignore rules.

Reply:
@JulienPalard i did some tests.
In the following lines you can see i'm printing the options of the `Similar` instance in the `_compute_sims` method.
As you can see, first two instances (one per process) have correct options (i.e those found in the pylintrc).
However those of the recombined object are not correct (see `self.ignore_imports`):
```
Options of Similar instance are: self.min_lines=40, self.ignore_comments=True, self.ignore_docstrings=True, self.ignore_imports=True, self.ignore_signatures=False
************* Module tests.unittest.pywbem.test_tupleparse
tests/unittest/pywbem/test_tupleparse.py:8674:9: W0511: TODO 1/18 AM: Should this not be rejected as invalid type? (fixme)
tests/unittest/pywbem/test_tupleparse.py:7:0: E0401: Unable to import 'pytest' (import-error)
tests/unittest/pywbem/test_tupleparse.py:8:0: E0611: No name 'version' in module 'packaging' (no-name-in-module)
tests/unittest/pywbem/test_tupleparse.py:8:0: E0401: Unable to import 'packaging.version' (import-error)
Options of Similar instance are: self.min_lines=40, self.ignore_comments=True, self.ignore_docstrings=True, self.ignore_imports=True, self.ignore_signatures=False
************* Module tests.unittest.pywbem.test_cim_obj
tests/unittest/pywbem/test_cim_obj.py:11931:5: W0511: TODO: Decide whether mutable child objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:11952:5: W0511: TODO: Decide whether value objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:13742:9: W0511: TODO: Clarify whether invalid embedded_object should be checked (fixme)
tests/unittest/pywbem/test_cim_obj.py:18820:5: W0511: TODO: Decide whether mutable child objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:18833:5: W0511: TODO: Decide whether value objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:22940:9: W0511: TODO 01/18 AM Enable test case once char16 produces single quotes (fixme)
tests/unittest/pywbem/test_cim_obj.py:30740:5: W0511: TODO: Decide whether mutable child objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:30785:5: W0511: TODO: Decide whether value objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:32625:9: W0511: TODO: Clarify whether invalid embedded_object should be checked (fixme)
tests/unittest/pywbem/test_cim_obj.py:37828:5: W0511: TODO: Decide whether mutable child objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:37841:5: W0511: TODO: Decide whether value objects should be copied, and add test: (fixme)
tests/unittest/pywbem/test_cim_obj.py:43508:28: W0511: TODO 01/18 AM Enable once char16 uses single quotes (fixme)
tests/unittest/pywbem/test_cim_obj.py:20:0: E0401: Unable to import 'pytest' (import-error)
tests/unittest/pywbem/test_cim_obj.py:22:0: E0611: No name 'version' in module 'packaging' (no-name-in-module)
tests/unittest/pywbem/test_cim_obj.py:22:0: E0401: Unable to import 'packaging.version' (import-error)
Recombining!
Options of Similar instance are: self.min_lines=40, self.ignore_comments=True, self.ignore_docstrings=True, self.ignore_imports=False, self.ignore_signatures=False
```
With the lines i added, then the recombined object has correct options.

Reply:
Do your implementation need those options during the recombining phase? In the current main branch, the recombining phase only use min_lines.

But I have nothing against passing too much options, it won't break anything.